docs/serving/samples/hello-world/helloworld-go/Dockerfile

```diff
@@ -1,28 +1,30 @@
-# Use the official Golang image to create a build artifact.
+# Use the official golang image to create a binary.
 # This is based on Debian and sets the GOPATH to /go.
 # https://hub.docker.com/_/golang
-FROM golang:1.13 as builder
+FROM golang:1.13-buster as builder
 
 # Create and change to the app directory.
 WORKDIR /app
 
-# Retrieve application dependencies using go modules.
-# Allows container builds to reuse downloaded dependencies.
+# Retrieve application dependencies.
+# This allows the container build to reuse cached dependencies.
+# Expecting to copy go.mod and if present go.sum.
 COPY go.* ./
 RUN go mod download
 
 # Copy local code to the container image.
 COPY . ./
 
 # Build the binary.
-# -mod=readonly ensures immutable go.mod and go.sum in container builds.
-RUN CGO_ENABLED=0 GOOS=linux go build -mod=readonly -v -o server
+RUN go build -mod=readonly -v -o server
```

Comment:
One of the main reasons Go binaries fail on Alpine is they reference glibc, which most commonly creeps in through `net/http`.  You are dropping `CGO_ENABLED=0` here, which directs go to link against a pure-Go version of `net/http`.

I am curious what issues we're looking to address here, so I can better under the what/why.